### DISCUSSION TITLE
关于项目二——地图的导入导出

### DISCUSSION THREAD

**Discussion by CSCSX** [liusy58/NJU_AdvancedProgramming_SP22#166] [Q&A] [answered]
请问地图的编辑是否需要GUI，即用户可以在可视化窗口中创作新的地图，然后导出。还是只允许对txt格式的资源文件地图编辑？如果支持用户对txt格式地图的修改，是否假定用户不会写出非法的语法？验收的时候是验收发布版还是debug版？如果验收发布版，用户无法在发布文件夹中找到原始的资源文件，那么创作地图的功能只能用GUI接口实现，而这似乎是个工作量不小的模块。求解答

**Comment by liusy58** [comment, marked as answer]
> 请问地图的编辑是否需要GUI，即用户可以在可视化窗口中创作新的地图，然后导出。还是只允许对txt格式的资源文件地图编辑？

不强制要求GUI，事实上你可以按照任何你觉得舒服的方式来。



> 如果支持用户对txt格式地图的修改，是否假定用户不会写出非法的语法？

我认为这里的用户是你自己，你应该能够给出正确的地图，这个只是一个小的模块，我认为首先build出来一个可以执行的东西之后在讨论各种cornel case会比你一开始就考虑的好。

> 验收的时候是验收发布版还是debug版？

验收的时候给出游戏的完整执行过程即可，不限制发布版还是debug版。

> 用户无法在发布文件夹中找到原始的资源文件

你可以直接在发布版里面预先准备好需要的各种资源。

**Comment by nayutatata** [comment]
为什么近战兵能打飞行单位呢（）
